examples/step-87/step-87.cc

```diff
@@ -0,0 +1,1175 @@
+ * The full text of the license can be found in the file LICENSE.md at
+ * the top level directory of deal.II.
+ *
+ * ---------------------------------------------------------------------
+ *
+ *
+ * Authors: Magdalena Schreter-Fleischhacker, Technical University of
+ *          Munich, 2023
+ *          Peter Munch, University of Augsburg, 2023
+ */
+
+// @sect3{Include files}
+//
+// The program starts with including all the relevant header files.
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/function_lib.h>
+#include <deal.II/base/function_signed_distance.h>
+#include <deal.II/base/mpi.h>
+#include <deal.II/base/mpi.templates.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/dofs/dof_renumbering.h>
+
+#include <deal.II/fe/fe_nothing.h>
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/mapping_q1.h>
+#include <deal.II/fe/mapping_q_cache.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <deal.II/lac/generic_linear_algebra.h>
+
+#include <deal.II/numerics/data_out.h>
+
+#include <iostream>
+#include <fstream>
+
+// The files most relevant for this tutorial are the ones that
+// contain Utilities::MPI::RemotePointEvaluation and the distributed evaluation
+// functions in the VectorTools namespace, which use
+// Utilities::MPI::RemotePointEvaluation.
+#include <deal.II/base/mpi_remote_point_evaluation.h>
+#include <deal.II/numerics/vector_tools.h>
+
+// The following header file provides the class FEPointEvaluation, which allows
+// us to evaluate values of a local solution vector at arbitrary unit points of
+// a cell.
+#include <deal.II/matrix_free/fe_point_evaluation.h>
+
+// In the following, we define utility functions that are used
+// below in the mini examples.
+
+// We pack everything that is specific for this program into a namespace
+// of its own.
+namespace Step87
+{
+  using namespace dealii;
+
+// The minimum requirement of this tutorial is MPI. If deal.II is built
+// with p4est, we use parallel::distributed::Triangulation as
+// distributed mesh. The class parallel::shared::Triangulation is
+// used if deal.II is built without p4est or if the dimension of the
+// triangulation is 1D, e.g., in the case of codim-1 meshes.
+#ifdef DEAL_II_WITH_P4EST
+  template <int dim, int spacedim = dim>
+  using DistributedTriangulation = typename std::conditional<
+    dim == 1,
+    parallel::shared::Triangulation<dim, spacedim>,
+    parallel::distributed::Triangulation<dim, spacedim>>::type;
+#else
+  template <int dim, int spacedim = dim>
+  using DistributedTriangulation =
+    parallel::shared::Triangulation<dim, spacedim>;
+#endif
+
+  // A list of points along a line is created by definition of a
+  // start point @p p0, an end point @p p1, and the number of subdivisions
+  // @p n_subdivisions.
+  template <int spacedim>
+  std::vector<Point<spacedim>>
+  create_points_along_line(const Point<spacedim> &p0,
```

Comment:
It feels a bit weird to start the tutorial with many of these small helper functions which are greatly useful to this tutorial. I also feel these helper functions could be used elsewhere in user codes. Do you think it might be a viable alternative to introduce them as utilities in the main deal.II library or you would rather have them reside there?

I feel it takes a bit of a time until we get to the "core" of the example itself.

Reply:
We were discussing that also during the development of this tutorial. For starting the discussion, I list the functions together with suggestions for potentially appropriate locations within deal.II library:

- `create_points_along_line()` --> This could fit in `GeometricUtilities` https://dealii.org/developer/doxygen/deal.II/namespaceGeometricUtilities.html although this namespace seems to be almost empty.
- `print_along_line()` --> In my opinion, this function is too specific to become part of the library. 
- `collect_support_points()` and  `collect_support_points_with_narrow_band()` --> namespace DoFTools
-  `std::vector<Point<spacedim>> convert(
    const LinearAlgebra::distributed::Vector<double> &support_points_unrolled)` --> This could be a member function of the `Vector` class.

What do you think about that?

Reply:
We revisited this topic once again. Most of the functions are specific to the requirements of this tutorial and would need to be extended for general use cases. Furthermore, it's challenging to identify an appropriate place for integrating the point-related functionalities into the library. As a result, we have reached the following intermediate compromise solution: We have separated the utility functions into a declaration section at the beginning and their definitions at the end. This approach does not preclude the possibility of integrating these functions into the library in the future if they prove useful for other cases as well.